src/main/scala/schema/Column.scala

```diff
@@ -34,4 +44,46 @@ class Column(
     }
     s"$name$primaryKey $foreignKeyString"
   }
+
+  // determines how a primary key can be determined:
+  // strict or lenient
+  def isPrimaryKeyCandidate: Boolean = {
+    if (cli.strictPrimaryKeyChecking) {
+      isStrongPrimaryKey
+    } else {
+      canBePrimaryKey
+    }
+  }
+
+  // verify whether up until now all values have been increasing
+  // ==> good indicator for a primary key
+  def verifyIncreasingValuesOnChange(): Unit = {
+    // if values have not been increasing before, we don't need to check again
+    if (!areValuesIncreasing) return;
+
+    val valuesAreMonotonous = values.values.toSeq.sliding(2).forall {
+      case x :: y :: _ => x < y
+      case _           => true
+    }
+    if (valuesAreMonotonous && areValuesIncreasing) {
+      areValuesIncreasing = true
+    } else {
+      areValuesIncreasing = false
+    }
+  }
+
+  // indicates whether the values indicate a strong primary key and the values are always increasing
+  private def isStrongPrimaryKey: Boolean = {
+    canBePrimaryKey && areValuesIncreasing && isColumnNamePotentiallyPrimaryKey
+  }
+
+  // verify whether the name indicates a good primary key candidate
+  private def isColumnNamePotentiallyPrimaryKey = {
+    columnName.matches("(?i:.*id)") ||
+    columnName.matches("(?i:.*nr)") ||
+    columnName.matches("(?i:.*name)") ||
```

Comment:
I am not shure if "name" is a good primary key indicator but I agree with the rest.

Reply:
Yeah, sorry, I appear to have mistyped, and meant "key" instead of name. Good catch!
So I'd propose this:

```suggestion
    columnName.matches("(?i:.*key)") ||
```